.github/workflows/inspektor-gadget.yml

```diff
@@ -85,6 +87,13 @@ jobs:
             IMAGE_TAG="latest"
         fi
         echo IMAGE_TAG=$IMAGE_TAG >> $GITHUB_ENV
+    - name: Choose container repository
+      id: choose-repo
+      uses: ./.github/actions/choose-container-repo
+      with:
+        registry: ${{ env.REGISTRY }}
+        container-image-release: ${{ env.RELEASE_CONTAINER_REPO }}
+        container-image-dev: ${{ env.DEVELOPMENT_CONTAINER_REPO }}
```

Comment:
> I really want to keep this as simple as possible and I am not convinced than hiding things in env make it easier to understand.

I don't understand what is the concern here. We can have a step like similar to `Set IMAGE_TAG` but for `CONTAINER_REPO:. 

```
        if [[ ${{ github.ref }} == 'refs/heads/main' || ${{ github.ref }} == 'refs/tags/v'* ]]; then
          CONTAINER_REPO=$env.RELEASE_CONTAINER_REPO
        else
          CONTAINER_REPO=$env.DEVELOPMENT_CONTAINER_REPO
        fi
        echo CONTAINER_REPO=$CONTAINER_REPO >> $GITHUB_ENV
```

It's true that we have to repeat this many times, but we're doing the same for `Choose container repository` too, hence not difference. Does it make sense to you?

Reply:
My concern in the previous PR was that it was setting the environment variables inside a custom action. Instead, using the action's output makes things more clear. I think it would make things even better to move the image-tag computation inside the `choose-container-repo` and have:

```yaml
CONTAINER_REPO=${{ steps.new-fancy-name.outputs.container-repo }} IMAGE_TAGE=${{ steps.new-fancy-name.outputs.image-tag }} make $kubectl_gadget
```